File: sdks/java/src/main/java/com/puxbay/resources/StockTransfersResource.java
package com.puxbay.resources;

import com.puxbay.Puxbay;
import com.puxbay.exceptions.PuxbayException;
import com.puxbay.models.PaginatedResponse;
import com.puxbay.models.StockTransfer;

public class StockTransfersResource {
    private final Puxbay client;

    public StockTransfersResource(Puxbay client) {
        this.client = client;
    }

    public PaginatedResponse<StockTransfer> list(int page, String status) throws PuxbayException {
        String query = "stock-transfers/?page=" + page;
        if (status != null) query += "&status=" + status;
        return (PaginatedResponse<StockTransfer>) client.request("GET", query, null, PaginatedResponse.class);
    }

    public StockTransfer get(String transferId) throws PuxbayException {
        return client.request("GET", "stock-transfers/" + transferId + "/", null, StockTransfer.class);
    }

    public StockTransfer create(StockTransfer transfer) throws PuxbayException {
        return client.request("POST", "stock-transfers/", transfer, StockTransfer.class);
    }

    public StockTransfer complete(String transferId) throws PuxbayException {
        return client.request("POST", "stock-transfers/" + transferId + "/complete/", null, StockTransfer.class);
    }
}

File: sdks/kotlin/src/main/kotlin/com/puxbay/resources/StockTransfersResource.kt
package com.puxbay.resources

import com.puxbay.Puxbay
import com.puxbay.models.StockTransfer
import com.puxbay.models.PaginatedResponse
import io.ktor.http.HttpMethod

class StockTransfersResource(client: Puxbay) : BaseResource(client) {
    suspend fun list(page: Int = 1, status: String? = null): PaginatedResponse<StockTransfer> {
        val statusParam = if (status != null) "&status=$status" else ""
        return client.request(HttpMethod.Get, "stock-transfers/?page=$page$statusParam")
    }
    
    suspend fun get(transferId: String): StockTransfer =
        client.request(HttpMethod.Get, "stock-transfers/$transferId/")
    
    suspend fun create(transfer: StockTransfer): StockTransfer =
        client.request(HttpMethod.Post, "stock-transfers/", transfer)
    
    suspend fun complete(transferId: String): StockTransfer =
        client.request(HttpMethod.Post, "stock-transfers/$transferId/complete/")
}
